client/application_msgraph.go

```diff
@@ -0,0 +1,242 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/authorization/mgmt/2018-01-01-preview/authorization"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+var _ MSGraphClient = (*AppClient)(nil)
+
+type AppClient struct {
+	Client   authorization.BaseClient
+	GraphURI string
+}
+
+type listApplicationsResponse struct {
+	Value []ApplicationResult `json:"value"`
+}
+
+func GetGraphURI(env string) (string, error) {
+	switch strings.ToUpper(env) {
+	case "AZUREPUBLICCLOUD", "":
+		return "https://graph.microsoft.com", nil
+	case "AZUREUSGOVERNMENTCLOUD":
+		return "https://graph.microsoft.us", nil
+	case "AZUREGERMANCLOUD":
+		return "https://graph.microsoft.de", nil
+	case "AZURECHINACLOUD":
+		return "https://microsoftgraph.chinacloudapi.cn", nil
+	default:
+		return "", fmt.Errorf("environment '%s' unknown", env)
+	}
+}
+
+// NewMSGraphApplicationClient creates a new MS Graph Client
+// to allow interaction with the MS Graph v1.0 API.
+func NewMSGraphApplicationClient(userAgentExtension string, graphURI string, auth autorest.Authorizer) (*AppClient, error) {
+	client := authorization.NewWithBaseURI(graphURI, "")
```

Comment:
I think it'd be useful to add a comment here explaining that we're intentionally not providing the subscription ID since it's not needed for our usage of the client.

Reply:
Added a comment to document the lack of subscription ID